tests/test_external_account.py

```diff
@@ -67,6 +67,7 @@
 
 VALID_TOKEN_URLS = [
     "https://sts.googleapis.com",
+    "https://sts.mtls.googleapis.com",
```

Comment:
Probably worth it to add some test cases

Reply:
This should get included in this loop here https://github.com/googleapis/google-auth-library-python/blob/690b05509b7679b1d32fe71f36760d0f8c739268/tests/test_external_account.py#L342

Reply:
Let's add some invalid cases too